crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1898,7 +1912,11 @@ impl<'src> Parser<'src> {
         self.parse_comma_separated_list(RecoveryContextKind::ListElements, |parser| {
             elts.push(
                 parser
-                    .parse_star_expression_or_higher(AllowNamedExpression::Yes)
+                    .parse_named_expression_or_higher(
```

Comment:
There's a difference between the new and old parsing in that the previous implementation didn't unset the `INCLUDE_IN` flag because it was stored on the parser context. 

I think that's correct, considering that we're inside of a list. Did you verify that all call sites that now create a new context can omit setting the `INCLUDE_IN`?

Reply:
We did unset the `FOR_TARGET` flag which is equivalent to setting the `INCLUDE_IN`. Here, the default context sets the `INCLUDE_IN` flag. Refer: https://github.com/astral-sh/ruff/pull/11055/files#r1575593474